test/firtool/memory.fir

```diff
@@ -64,69 +64,27 @@ circuit Qux: %[[{
 ; multiple passes.  It should be replaced with a narrower test.
 
 ;CHECK-LABEL: hw.module @Qux
-;CHECK:    %[[memory_0:.+]] = sv.reg
-;CHECK:    %[[memory_1:.+]] = sv.reg
-;CHECK:    %[[memory_2:.+]] = sv.reg
-;CHECK:    %[[memory_3:.+]] = sv.reg
-;CHECK:    %[[addr:.+]] = sv.reg
-;CHECK:    %[[v4:.+]] = sv.read_inout %[[addr]]
-;CHECK:    %[[v5:.+]] = hw.array_create
-;CHECK:    %[[multbit_mux_wire:.+]] = sv.wire  : !hw.inout<i8>
-;CHECK:    %[[array_get:.+]] = hw.array_get %[[v5]][%[[v4]]]
-;CHECK:    sv.assign %[[multbit_mux_wire]], %[[array_get]]
-;CHECK:    %[[v6:.+]] = sv.read_inout %[[multbit_mux_wire]] : !hw.inout<i8>
-;CHECK:    %[[multbit_mux_wire:.+]] = sv.wire  : !hw.inout<i8>
-;CHECK:    %[[array_get:.+]] = hw.array_get %[[v5]][%rwAddr]
-;CHECK:    %[[v7:.+]] = sv.read_inout %[[multbit_mux_wire]] : !hw.inout<i8>
-;CHECK:    %12 = comb.icmp eq %rwAddr, %c0_i2 : i2
-;CHECK:    %13 = comb.and %rwEn, %rwMode, %rwMask, %12 : i1
-;CHECK:    %14 = comb.icmp bin eq %rwAddr, %c1_i2 : i2
-;CHECK:    %15 = comb.and %rwEn, %rwMode, %rwMask, %14 : i1
-;CHECK:    %16 = comb.icmp bin eq %rwAddr, %c-2_i2 : i2
-;CHECK:    %17 = comb.and %rwEn, %rwMode, %rwMask, %16 : i1
-;CHECK:    %18 = comb.icmp bin eq %rwAddr, %c-1_i2 : i2
-;CHECK:    %19 = comb.and %rwEn, %rwMode, %rwMask, %18 : i1
-;CHECK:    %20 = comb.icmp eq %wAddr, %c0_i2 : i2
-;CHECK:    %21 = comb.and %wEn, %wMask, %20 : i1
-;CHECK:    %22 = comb.icmp bin eq %wAddr, %c1_i2 : i2
-;CHECK:    %23 = comb.and %wEn, %wMask, %22 : i1
-;CHECK:    %24 = comb.icmp bin eq %wAddr, %c-2_i2 : i2
-;CHECK:    %25 = comb.and %wEn, %wMask, %24 : i1
-;CHECK:    %26 = comb.icmp bin eq %wAddr, %c-1_i2 : i2
-;CHECK:    %27 = comb.and %wEn, %wMask, %26 : i1
+;CHECK:    %memory = sv.reg  : !hw.inout<array<4xi8>>
+;CHECK:    %[[v0:.+]] = sv.read_inout %memory : !hw.inout<array<4xi8>>
+;CHECK:    %addr = sv.reg  : !hw.inout<i2>
+;CHECK:    %[[v1:.+]] = sv.read_inout %addr : !hw.inout<i2>
+;CHECK:    %[[v2:.+]] = hw.array_get %[[v0]][%1] : !hw.array<4xi8>
+;CHECK:    %[[v3:.+]] = hw.array_get %[[v0]][%rwAddr] : !hw.array<4xi8>
+;CHECK:    %[[v4:.+]] = comb.icmp eq %wAddr, %c0_i2 : i2
+;CHECK:    %[[v5:.+]] = comb.and %wEn, %wMask, %[[v4]] : i1
+;CHECK:    %[[v6:.+]] = comb.mux bin %[[v5]], %wData, %rwDataIn : i8
+;CHECK:    %[[v7:.+]] = comb.icmp bin eq %wAddr, %c1_i2 : i2
+;CHECK:    %[[v8:.+]] = comb.and %wEn, %wMask, %[[v7]] : i1
+;CHECK:    %[[v9:.+]] = comb.mux bin %[[v8]], %wData, %rwDataIn : i8
+;CHECK:    %[[v10:.+]] = comb.icmp bin eq %wAddr, %c-2_i2 : i2
+;CHECK:    %[[v11:.+]] = comb.and %wEn, %wMask, %[[v10]] : i1
+;CHECK:    %[[v12:.+]] = comb.mux bin %[[v11]], %wData, %rwDataIn : i8
+;CHECK:    %[[v13:.+]] = comb.icmp bin eq %wAddr, %c-1_i2 : i2
+;CHECK:    %[[v14:.+]] = comb.and %wEn, %wMask, %[[v13]] : i1
+;CHECK:    %[[v15:.+]] = comb.mux bin %[[v14]], %wData, %rwDataIn : i8
+;CHECK:    %[[v16:.+]] = hw.array_create %[[v15]], %[[v12]], %[[v9]], %[[v6]] : i8
```

Comment:
Not directly related to the PR, this is the result of memory.fir. 
```verilog
   memory <= {{wEn & wMask & (&wAddr) ? wData : rwDataIn}, {wEn & wMask & wAddr == 2'h2 ? wData :
                                rwDataIn}, {wEn & wMask & wAddr == 2'h1 ? wData : rwDataIn}, {wEn & wMask & wAddr == 2'h0 ?
                                wData : rwDataIn}};
```
 I think we have to modify LowerSeqToSV to avoid emitting verilog like this.

Reply:
Also it looks there is a bug somewhere `memory` must be appear in the rhs.